rs/ic_os/config/src/guest_vm_config.rs

```diff
@@ -0,0 +1,457 @@
+    fn test_make_bootstrap_options() {
+        let mut config = create_test_hostos_config();
+        config.icos_settings.use_nns_public_key = true;
+        config.icos_settings.use_ssh_authorized_keys = true;
+        config.icos_settings.use_node_operator_private_key = true;
+
+        let guestos_config = generate_guestos_config(&config).unwrap();
+
+        let options =
+            make_bootstrap_options(&config, guestos_config, Path::new("/tmp/test")).unwrap();
+
+        assert_eq!(
+            options,
+            BootstrapOptions {
+                ipv6_address: Some("2001:db8::6801:aeff:fe1a:9bb/64".to_string()),
+                ipv6_gateway: Some("2001:db8::ffff".to_string()),
+                ipv4_address: Some("192.168.1.2/24".to_string()),
+                ipv4_gateway: Some("192.168.1.1".to_string()),
+                domain: Some("test.domain".to_string()),
+                node_reward_type: Some("type3.1".to_string()),
+                hostname: Some("guest-001122334455".to_string()),
+                nns_urls: vec!["https://example.com/".to_string()],
+                guestos_config: Some(PathBuf::from("/tmp/test")),
+                nns_public_key: Some(PathBuf::from("/boot/config/nns_public_key.pem")),
+                node_operator_private_key: Some(PathBuf::from(
+                    "/boot/config/node_operator_private_key.pem"
+                )),
+                accounts_ssh_authorized_keys: Some(PathBuf::from(
+                    "/boot/config/ssh_authorized_keys"
+                )),
+                ..Default::default()
+            }
+        );
+    }
+
+    fn goldenfiles_path() -> PathBuf {
+        let mut path = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+        path.push("golden");
+        path
+    }
+
+    #[test]
+    fn test_generate_vm_config_qemu() {
+        let mut mint = Mint::new(goldenfiles_path());
+        let mut config = create_test_hostos_config();
+
+        config.hostos_settings = HostOSSettings {
+            vm_memory: 490,
+            vm_cpu: "qemu".to_string(),
+            vm_nr_of_vcpus: 56,
+            verbose: true,
+        };
+
+        let vm_config = generate_vm_config(&config, Path::new("/tmp/config.img")).unwrap();
+
+        fs::write(
+            mint.new_goldenpath("guestos_vm_qemu.xml").unwrap(),
+            vm_config,
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn test_generate_vm_config_kvm() {
+        let mut mint = Mint::new(goldenfiles_path());
+        let mut config = create_test_hostos_config();
+        config.hostos_settings = HostOSSettings {
+            vm_memory: 490,
+            vm_cpu: "kvm".to_string(),
+            vm_nr_of_vcpus: 56,
+            verbose: false,
+        };
+
+        let vm_config = generate_vm_config(&config, Path::new("/tmp/config.img")).unwrap();
+
+        fs::write(
+            mint.new_goldenpath("guestos_vm_kvm.xml").unwrap(),
+            vm_config,
+        )
+        .unwrap();
+    }
```

Comment:
```suggestion
fn test_vm_config(cpu_type: &str, filename: &str) {
    let mut mint = Mint::new(goldenfiles_path());
    let mut config = create_test_hostos_config();
    
    config.hostos_settings = HostOSSettings {
        vm_memory: 490,
        vm_cpu: cpu_type.to_string(),
        vm_nr_of_vcpus: 56,
        verbose: false,
    };

    let vm_config = generate_vm_config(&config, Path::new("/tmp/config.img")).unwrap();
    fs::write(mint.new_goldenpath(filename).unwrap(), vm_config).unwrap();
}

#[test]
fn test_generate_vm_config_qemu() {
    test_vm_config("qemu", "guestos_vm_qemu.xml");
}

#[test]
fn test_generate_vm_config_kvm() {
    test_vm_config("kvm", "guestos_vm_kvm.xml");
}
```

You would just need to also fix the qemu golden file to have the verbose flag be false